ifrs17/Import/Importers.ipynb

```diff
@@ -1106,31 +1145,29 @@
             "source": [
                 "Import.DefineFormat(ImportFormats.Actual, async (options, dataSet) => {",
                 "\n    Activity.Start();",
-                "\n    var args = (await GetArgsFromMainAsync<PartitionByReportingNodeAndPeriod>(dataSet)) with {ImportFormat = ImportFormats.Actual};",
-                "\n    await DataNodeFactoryAsync(dataSet, ImportFormats.Actual, args);",
+                "\n    var mainArgs = (await GetArgsFromMainAsync<PartitionByReportingNodeAndPeriod>(dataSet)) with {ImportFormat = ImportFormats.Actual};",
                 "\n    if(Activity.HasErrors()) return Activity.Finish();",
                 "\n",
-                "\n    var workspace = Workspace.CreateNew();",
-                "\n    var parsingLog = await ParseActualsToWorkspaceAsync(dataSet, args, workspace);",
-                "\n    if(parsingLog.Errors.Any()) return Activity.Finish().Merge(parsingLog);",
-                "\n",
-                "\n    var storage = new ImportStorage(args, DataSource, workspace);",
-                "\n    await storage.InitializeAsync();",
-                "\n    if(Activity.HasErrors()) return Activity.Finish().Merge(parsingLog);",
+                "\n    var targetArgs = mainArgs.Scenario == null ",
+                "\n                     ? mainArgs.RepeatOnce().Concat(await DataSource.Query<PartitionByReportingNodeAndPeriod>() ",
+                "\n                               .Where(x => x.ReportingNode == mainArgs.ReportingNode && x.Year == mainArgs.Year && x.Month == mainArgs.Month && x.Scenario != null)",
+                "\n                               .Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArrayAsync())",
+                "\n                     : mainArgs.RepeatOnce();",
                 "\n",
-                "\n    var universe = Scopes.ForStorage(storage).ToScope<IModel>();",
-                "\n    var identities = universe.GetScopes<GetIdentities>(storage.DataNodesByImportScope[ImportScope.Primary]).SelectMany(s => s.Identities);",
-                "\n    var ivs = universe.GetScopes<ComputeIfrsVarsActuals>(identities).SelectMany(x => x.CalculatedIfrsVariables);",
-                "\n    if(Activity.HasErrors()) return Activity.Finish().Merge(parsingLog);",
+                "\n    if (targetArgs.Count() > 1) ApplicationMessage.Log(Warning.ScenarioReCalculations, String.Join(\", \", targetArgs.Skip(1).Select(x => x.Scenario)));",
```

Comment:
Same as above

Reply:
So you need this code repetition?

Reply:
I can implement a refactoring in the next PR since I need to implement it also in YieldCurve Importer PR

Reply:
implemented